tests/nn/data_parallel/test_sharded_ddp.py

```diff
@@ -145,55 +171,29 @@ def check_parity(amp: bool):
             module=model, sharded_optimizer=sharded_optimizer, broadcast_buffers=True
         )
 
+        next(model.parameters()).requires_grad = False  # Test non-trainable parameters
```

Comment:
the whole point of this PR.. check that you can change the training graph after instantiation and still get the correct results